packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -402,9 +399,34 @@ function useRestoreFocus(scopeRef: RefObject<Element[]>, restoreFocus: boolean,
   // create a ref during render instead of useLayoutEffect so the active element is saved before a child with autoFocus=true mounts.
   const nodeToRestoreRef = useRef(typeof document !== 'undefined' ? document.activeElement as FocusableElement : null);
 
+  // restoring scopes should all track if they are active regardless of contain, but contain already tracks it plus logic to contain the focus
+  // restoring-non-containing scopes should only care if they become active so they can perform the restore
+  useLayoutEffect(() => {
+    let scope = scopeRef.current;
+    if (!restoreFocus || contain) {
```

Comment:
hmm, yeah. it would need to check if it is the inner-most scope with restoreFocus.

reason I was thinking was if a focus scope started out with both contain and restoreFocus disabled, and then got one of them enabled later, it wouldn't become the active scope if it already contained focus.

but if it's a big change, then we could do it separately (or when it comes up)

Reply:
yeah, i mean, this is also an issue when you have
```
<FocusScope restoreFocus>
  <FocusScope contain />
</FocusScope>
```

so maybe I should handle it here, but am happy to do follow up
maybe if an active focusscope unmounts, if it's not a restoring focusscope, it marks its parent as the active scope?